src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue

```diff
@@ -488,6 +495,32 @@ export default {
 			}
 		},
 
+		/**
+		 * Attributes to apply on the root node
+		 * This always includes the `class` and `style` attributes, for button style everything is applied.
+		 * For checkbox / radio only `class` and `style` are applied - all other are forwarded to the input element.
+		 */
+		rootAttributes() {
+			if (this.isButtonType) {
+				return this.$attrs
+			} else {
+				return {
+					class: this.$attrs.class,
+					style: this.$attrs.style,
+				}
+			}
+		},
+
+		/**
+		 * When not button type all attributes (except `style` and `class`) are forwarded to the input element
+		 */
+		inputAttributes() {
+			const attributes = { ...this.$attrs }
+			delete attributes.class
+			delete attributes.style
+			return attributes
+		},
```

Comment:
Though it works in Options API, `$attrs` is not a really reactive object and should not be used in computed.

To pass `class` and `style` to the root, we can do the same we did with other components in `@nextcluod/vue@9` already - explicitly define them as props and set on the element. This would also allow to ger rid of coping attrs object

Reply:
Makes sense, fixed this but needed a `computed` for the class. Otherwise there are error because "class" is a reserved word so `[class, { ... }]` is not valid.

Reply:
Props are also available directly in template as `$props.class`

https://github.com/nextcloud-libraries/nextcloud-vue/blob/next/src/components/NcInputField/NcInputField.vue#L36